# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1252907715 -7200
#      Mon Sep 14 07:55:15 2009 +0200
# Branch release-3-2-x
# Node ID f12b581a157283183ab91b8ad35c226af8f0f7ef
# Parent  f859b61b8bf4dbfd1328ae7bb873c6c853bbce78
gnuplot_drawnow.m: Avoid flickering plot windows.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2009-09-09  Tatsuro Matsuoka <tmacchant@yahoo.co.jp>
+
+	* plot/gnuplot_drawnow.m: Avoid flickering windows by avoding
+	'set multiplot' / 'unset multiplot' for gnuplot's windows and
+	wxt terminals.
+
 2009-09-08  John W. Eaton  <jwe@octave.org>
 
 	* io/dlmwrite.m: Fix typo.
 
 2009-09-05  John W. Eaton  <jwe@octave.org>
 
 	* plot/hold.m: Correctly toggle hold state.  Set both figure and
 	axes "nextplot" property when turning hold state on.
diff --git a/scripts/plot/gnuplot_drawnow.m b/scripts/plot/gnuplot_drawnow.m
--- a/scripts/plot/gnuplot_drawnow.m
+++ b/scripts/plot/gnuplot_drawnow.m
@@ -281,30 +281,32 @@ function [enhanced, implicit_margin] = g
       ## Options must go last.
       term_str = sprintf ("%s %s", term_str, opts_str);
     endif
     if (! isempty (size_str) && new_stream)
       ## size_str comes after other options to permit specification of
       ## the canvas size for terminals cdr/corel.
       term_str = sprintf ("%s %s", term_str, size_str);
     endif
-    ## Work around the gnuplot feature of growing the x11 window when
-    ## the mouse and multiplot are set.
+    ## Work around the gnuplot feature of growing the x11 window and
+    ## flickering window (x11, windows, & wxt) when the mouse and
+    ## multiplot are set in gnuplot.
     fputs (plot_stream, "unset multiplot;\n");
-    if (! strcmp (term, "x11")
+    flickering_terms = {"x11", "windows", "wxt"};
+    if (! any (strcmp (term, flickering_terms))
         || numel (findall (h, "type", "axes")) > 1
         || numel (findall (h, "type", "image")) > 0)
       fprintf (plot_stream, "%s\n", term_str);
       if (nargin == 5)
         if (! isempty (file))
           fprintf (plot_stream, "set output '%s';\n", file);
         endif
       endif
       fputs (plot_stream, "set multiplot;\n");
-    elseif (strcmp (term, "x11"))
+    elseif (any (strcmp (term, flickering_terms)))
       fprintf (plot_stream, "%s\n", term_str);
       if (nargin == 5)
         if (! isempty (file))
           fprintf (plot_stream, "set output '%s';\n", file);
         endif
       endif
     endif
   else
